divi/src/BlockMemoryPoolTransactionCollector.cpp

```diff
@@ -103,7 +103,8 @@ BlockMemoryPoolTransactionCollector::BlockMemoryPoolTransactionCollector(
 
 bool BlockMemoryPoolTransactionCollector::VerifyUTXOIsKnownToMemPool (const CTxIn& txin, bool& fMissingInputs)  const
 {
-    if(!mempool_.mapTx.count(txin.prevout.hash)){
+    CTransaction tx;
+    if(!mempool_.lookup(txin.prevout.hash, tx)) {
         LogPrintf("ERROR: mempool transaction missing input\n");
```

Comment:
This change and the change inside *::PrioritizeMempoolTransactions could be combined into part of the same function to avoid doing a double lookup. Refactor VerifyUTXOIsKnownToMemPool to CollectFundingTransactionFromMempool and use the recovered transaction to access the value.

Reply:
or some similarly verbose and clear name :P

Reply:
Sounds good.  The lookup is quite cheap (as it is just an in-memory `std::map` lookup), but still makes a lot of sense to me.  I'll add this in.